### PR TITLE
Lint enhancements

### PR DESCRIPTION
**Pull request by mike-singh**
Please **test locally** before merging. 

1) Introduces rulesets from various libraries

```   
   "tslint-consistent-codestyle": "^1.12.2",
    "tslint-immutable": "^4.5.2",
    "tslint-microsoft-contrib": "^5.0.3",
    "tslint-sonarts": "^1.6.0"
```
as well as ```tslint:all```

This is an upstream merge to the parent fork; this will need to be merged into our master after testing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,8 +21,8 @@
     "runjs": "^4.3.0"
   },
   "scripts": {
-    "test": "run test",
-    "postinstall": "node install"
```
Comment (jasonmorganson) [inline package.json]: This script needs to stay.

**Comment by jasonmorganson** [comment]
A commit from the downstream CredSimple repo snuck in. You want to rebase this and remove https://github.com/jasonmorganson/run-scripts/pull/2/commits/240cdec889f45df1f3e8489ba1a5670f51260031

File: tslint.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-    "extends": [
-        "tslint:all"
-    ],
-    "rules": {
```
[the remark this answers is not in the document]
Comment (mike-singh) [reply tslint.json]: no - https://github.com/jasonmorganson/run-scripts/pull/2/files#diff-cdc212ffa7f199f29da017856c604779R20

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@jasonmorganson/run-scripts",
-  "version": "1.5.1",
+  "version": "1.5.2",
```
[the remark this answers is not in the document]
Comment (mike-singh) [reply package.json]: No need - you're bumping to 1.6 soon in your other branch

**Pull request merged**